src/test/java/com/salesforce/dynamodbv2/UpdateTest.java

```diff
@@ -89,55 +92,230 @@ void update(TestArgument testArgument) {
     @ParameterizedTest
     @ArgumentsSource(DefaultArgumentProvider.class)
     @DefaultArgumentProviderConfig(tables = { TABLE1 })
-    void updatePrimaryKeyFieldFails(TestArgument testArgument) {
+    void addUpdate(TestArgument testArgument) {
         testArgument.forEachOrgContext(org -> {
             Map<String, AttributeValue> updateItemKey = ItemBuilder.builder(testArgument.getHashKeyAttrType(),
                 HASH_KEY_VALUE)
                 .build();
             UpdateItemRequest updateItemRequest = new UpdateItemRequest()
                 .withTableName(TABLE1)
                 .withKey(updateItemKey)
-                .withUpdateExpression("set #someField = :someValue")
-                .withExpressionAttributeNames(ImmutableMap.of("#someField", HASH_KEY_FIELD))
+                .withUpdateExpression("add #someOtherField :" + SOME_OTHER_FIELD_VALUE)
+                .withExpressionAttributeNames(ImmutableMap.of("#someOtherField","someOtherField"))
+                .withExpressionAttributeValues(ImmutableMap.of(":" + SOME_OTHER_FIELD_VALUE,
+                    createNumberAttribute("1")));
```

Comment:
We should also have a case for when there's already an existing value on an N attribute, and then we do an update on the attribute and ADD on top of the existing value.

Reply:
Also, to make sure this works when we're ADDing to an attribute that's part of a secondary index, we should have a test case like updateGsiFieldAndQueryGsi()

Reply:
Yeah these are the test cases I hadn't updated yet after running into the validation mentioned above: runUpdateConditionalOnGsiRkTest, runUpdateHkRkTableTest, updateGsiFieldAndQueryGsi, updateFieldInMultipleGsis.